### ISSUE TITLE
Implement --cpus mapped to QEMU -smp arguments

### ISSUE THREAD

**Issue by grahamwhaley** [clearcontainers/runtime#393]
We should map the `docker run --cpus` request do the QEMU VM command line `-smp` argument, as best we can.
It cannot be a 1:1 mapping, as `--cpus` takes a float argument, and implements partial CPU quotas using the CFS period and quota settings for instance.
A proposal is that we round up to the nearest whole CPU value, for the act of 'least surprise'.

Also see Issue #341 for more background and info, and how this relates in the future to the setup of POD cpu constraints.

**Comment by jodh-intel** [comment]
Whatever we decide, we must also ensure https://github.com/clearcontainers/runtime/blob/master/docs/limitations.md gets updated with the details.

**Comment by grahamwhaley** [comment]
Ah, indeed, I meant to mention that - once we have implemented this we need to describe at least the non 1:1 mapping limitation on `--cpus`, and also probably call out more specifically the other things we don't support (quoata and slice things etc.)

**Comment by wcwxyz** [comment]
I was thinking, maybe when --cpus 1.5 is requested, we should just fail instead of round up to 2. So user will not be confused or misled.

How about we support --cpus int as the first step and fix --cpu float in the future if we have better idea how to handle it.

**Comment by grahamwhaley** [comment]
Yep, I'm happy with that as a first step @wcwxyz Better than no --cpus support at all :-)

**Comment by wcwxyz** [comment]
I'd like to get https://github.com/clearcontainers/runtime/pull/390 merged before I start working on this one. This `docker --cpu CPU` support is pretty like the `docker -m MEMORY` one. They can share the same code in some way.
